addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,297 @@
+                    "ItemAmount": round(line.price_total, 2),
+                }
+            )
+        return res
+
+    def _l10n_tw_edi_generate_invoice_json(self):
+        if not self.env.company.l10n_tw_edi_ecpay_merchant_id:
+            raise UserError(_("Please fill in the ECpay API information in the Setting!"))
+
+        # prepare json data
+        self.l10n_tw_edi_related_number = base64.urlsafe_b64encode(uuid.uuid4().bytes).decode()
+        json_data = {
+            "MerchantID": self.env.company.l10n_tw_edi_ecpay_merchant_id,
+            "RelateNumber": self.l10n_tw_edi_related_number,
+            "CustomerIdentifier": self.l10n_tw_edi_customer_identifier or "",
+            "CustomerName": self.l10n_tw_edi_customer_name,
+            "CustomerAddr": self.l10n_tw_edi_customer_address,
+            "CustomerPhone": self.l10n_tw_edi_customer_phone or '',
+            "CustomerEmail": self.l10n_tw_edi_customer_email or '',
+            "ClearanceMark": "",
+            "Print": str(int(self.l10n_tw_edi_is_print or self.l10n_tw_edi_customer_identifier)),
+            "Donation": str(int(self.l10n_tw_edi_is_donate)),
+            "TaxType": "1",
+            "SalesAmount": int(self.amount_total),
+            "InvoiceRemark": "Odoo",
+            "InvType": "07",
+            "vat": self.l10n_tw_edi_ecpay_tax_type,
+            "CarrierType": self.l10n_tw_edi_carrier_type or "",
+            "CarrierNum": self.l10n_tw_edi_carrier_number if self.l10n_tw_edi_carrier_type in ["2", "3"] else "",
+            "Items": self._l10n_tw_edi_prepare_item_list()
+        }
+        if self.l10n_tw_edi_is_donate:
+            json_data.update({"LoveCode": self.l10n_tw_edi_love_code})
+        return json_data
+
+    def _transfer_time(self, time_before):
+        time_after = (
+            datetime.datetime.strptime(time_before, "%Y-%m-%d %H:%M:%S") - datetime.timedelta(hours=8)
+        ).strftime("%Y-%m-%d %H:%M:%S")
+        return time_after
+
+    def _l10n_tw_edi_send(self, json_content):
+        request_url, merchant_id, hashkey, hashIV = self.env.company._l10n_tw_edi_prepare_api_param()
+        # Ensure to lock the records that will be sent, to avoid risking sending them twice.
+        self.env["res.company"]._with_locked_records(self)
+        # prepare EcPayAPI class
+        response_data = EcPayAPI(hashkey, hashIV, merchant_id).call_ecpay_api(request_url + "/Issue", json_content)
+        if response_data.get("RtnCode") != 1:
+            return [_(f"Invoice {self.name} Error: {response_data.get('RtnMsg')}")]
+        self.l10n_tw_edi_ecpay_invoice_id = response_data.get("InvoiceNo")
+        self.l10n_tw_edi_invoice_create_date = self._transfer_time(response_data.get("InvoiceDate").replace("+", " "))
+        self.l10n_tw_edi_state = "invoiced"
+        self._message_log(
+            body=_("The invoice has been successfully sent to Ecpay with Ecpay invoice number '%s'.", response_data.get('InvoiceNo')),
```

Comment:
It's good practice to name your parameters in translated strings to ease the work of translators.
```suggestion
            body=_("The invoice has been successfully sent to Ecpay with Ecpay invoice number '%(invoice_number)s'.", invoice_number=response_data.get('InvoiceNo')),
```